lib/portlayer/exec/container.go

```diff
@@ -258,6 +266,16 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
 	// how widely it's used but couldn't - will ask around.
 	dsPath := fmt.Sprintf("[%s] %s", url.Host, url.Path)
 
+	state, err := c.vm.PowerState(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to get vm power status %q: %s", c.vm.Reference(), err)
+	}
+
+	// don't remove the containerVM if it is powered on
+	if state == types.VirtualMachinePowerStatePoweredOn {
+		return RemovePowerError{fmt.Errorf("You cannot remove a running container. Stop the container before attempting removal or use -f")}
```

Comment:
This error string is docker specific and should be in the docker personality. The port layer does not know what `-f` is. You should view the port layer as a library that can stand on its own, and so the errors should reflect that. Over here just "container is running," or something similar would suffice.